### ISSUE TITLE
Can't generate Models through VS

### ISSUE THREAD

**Issue by ad794** [zpqrtbnk/Zbu.ModelsBuilder#64]
Hi,

I keep getting the following error when I try to generate Models through VS by right clicking on the file. Any help would be greatly appreciated!

**_**ZbuModelsBuilder: Starting v2.1.3.52 13/08/2015 15:05:16.**_**
**_**ZbuModelsBuilder: ZbuModelsBuilder failed to generate code: Exception: Response status code does not indicate success (InternalServerError)**_**
**_**{"Message":"An error has occurred.","ExceptionMessage":"Value cannot be null.\r\nParameter name: v1","ExceptionType":"System.ArgumentNullException","StackTrace":"   at System.Version.op_LessThanOrEqual(Version v1, Version v2)\r\n   at Zbu.ModelsBuilder.Compatibility.IsCompatible(Version clientVersion)\r\n   at Zbu.ModelsBuilder.AspNet.ModelsBuilderApiController.CheckVersion(Version clientVersion, Version minServerVersionSupportingClient)\r\n   at Zbu.ModelsBuilder.AspNet.ModelsBuilderApiController.ValidateClientVersion(ValidateClientVersionData data)\r\n   at lambda_method(Closure , Object , Object[] )\r\n   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.<>c__DisplayClass13.<GetExecutor>b__c(Object instance, Object[] methodParameters)\r\n   at System.Threading.Tasks.TaskHelpers.RunSynchronously[TResult](Func`1 func, CancellationToken cancellationToken)"}**_**
**_**ZbuModelsBuilder:    at Zbu.ModelsBuilder.AspNet.ModelsBuilderApi.EnsureSuccess(HttpResponseMessage result)**_**
   **_**at Zbu.ModelsBuilder.AspNet.ModelsBuilderApi.ValidateClientVersion()**_**
   **_**at Zbu.ModelsBuilder.CustomTool.CustomTool.ZbuModelsBuilder.GenerateRaw(String wszInputFilePath, String wszDefaultNamespace, IntPtr[] rgbOutputFileContents, UInt32& pcbOutput, String& errMsg)**_**
**_**ZbuModelsBuilder: Starting v2.1.3.52 13/08/2015 15:12:43.**_**
**_**ZbuModelsBuilder: ZbuModelsBuilder failed to generate code: Exception: Response status code does not indicate success (InternalServerError)**_**
**_**{"Message":"An error has occurred.","ExceptionMessage":"Value cannot be null.\r\nParameter name: v1","ExceptionType":"System.ArgumentNullException","StackTrace":"   at System.Version.op_LessThanOrEqual(Version v1, Version v2)\r\n   at Zbu.ModelsBuilder.Compatibility.IsCompatible(Version clientVersion)\r\n   at Zbu.ModelsBuilder.AspNet.ModelsBuilderApiController.CheckVersion(Version clientVersion, Version minServerVersionSupportingClient)\r\n   at Zbu.ModelsBuilder.AspNet.ModelsBuilderApiController.ValidateClientVersion(ValidateClientVersionData data)\r\n   at lambda_method(Closure , Object , Object[] )\r\n   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.<>c__DisplayClass13.<GetExecutor>b__c(Object instance, Object[] methodParameters)\r\n   at System.Threading.Tasks.TaskHelpers.RunSynchronously[TResult](Func`1 func, CancellationToken cancellationToken)"}**_**
**_**ZbuModelsBuilder:    at Zbu.ModelsBuilder.AspNet.ModelsBuilderApi.EnsureSuccess(HttpResponseMessage result)**_**
   **_**at Zbu.ModelsBuilder.AspNet.ModelsBuilderApi.ValidateClientVersion()**_**
   **_**at Zbu.ModelsBuilder.CustomTool.CustomTool.ZbuModelsBuilder.GenerateRaw(String wszInputFilePath, String wszDefaultNamespace, IntPtr[] rgbOutputFileContents, UInt32& pcbOutput, String& errMsg)**_**

**Comment by ad794** [comment]
Also when I do Get-Package I see the following

Now I have tried my best to unistall 2.0.2-final but nuget unistall doesn;t seem to find that.

Zbu.ModelsBuilder              2.0.2-final          ZpqrtBnk Umbraco ModeslBuilder Libraries.                                                                               
Zbu.ModelsBuilder              2.1.3                ZpqrtBnk Umbraco ModeslBuilder Libraries.                                                                               
Zbu.ModelsBuilder.AspNet       2.0.2-final          ZpqrtBnk Umbraco ModelsBuilder Site Integration.                                                                        
Zbu.ModelsBuilder.AspNet       2.1.3                ZpqrtBnk Umbraco ModelsBuilder Site Integration.

**Comment by zpqrtbnk** [comment]
From the stack trace, it looks like you are running version v2.1.3.52 of the VSIX extension, ie you updated the VSIX extension... but may be running an older version of the web component. Too old to properly report that versions are not compatible. Can you check and report the version of the Zbu.ModelsBuilder.dll file in your ~/bin directory?

**Comment by ad794** [comment]
Hi Stephan,

Thanks you for your reply! Below is a screen shot showing the dll version I am using

Many thanks for your help again!


![dll](https://cloud.githubusercontent.com/assets/6419446/9497169/a3faa456-4c0b-11e5-8a89-b37a1ce970d6.PNG)

**Comment by zpqrtbnk** [comment]
You are using version 2.0.2.31 of the web component, and the VSIX extension cannot talk to that version. You also need to update the web site (using NuGet, Umbraco package... whichever method was used in the first place).

**Comment by ad794** [comment]
Thanks for your help. 

For some reason VS was still picking up the old 2.0.2-final version even when I removed and refrenced the one. I had to Force Uninstall eveything using NuGet.

And then Istall 2.1.3 by specifying explicity -Version 2.1.3 in Nuget Install statement.

Now everything is working can;t believe I was stuck on this for days but glad its finally sorted. Really really appreciate your help!!

Have a good day

Many thanks
Adnan
